quill-core/src/main/scala/io/getquill/ast/Ast.scala

```diff
@@ -104,4 +106,11 @@ case class Assignment(input: Ident, property: String, value: Ast)
 
 //************************************************************
 
+sealed trait SchemaDefinition extends Ast
```

Comment:
I think it's not necessary, you could do something like:

Parsing.scala
```scala
case q"$pack.query[${ t: Type }].apply($config)" =>
      Entity(schemaConfigParsing(config))
```

SchemaConfigParsing.scala
```scala
case class Config(table: Option[String] = None, generated: List[Property] = List())

object SchemaConfigParsing {
   def apply(t: Tree): Config =
     tree match {
         case "$c.table(${ name: String })" =>
              apply(c).copy(table = name)
         case "$c.generated(..$properties)" =>
              apply(c).copy(generated = ...)
         case Ident(_) =>
             Config()
     }
}
```